artemis/src/main/java/com/artemis/InvocationStrategy.java

```diff
@@ -7,11 +7,10 @@
 	protected void process(Bag<BaseSystem> systems) {
 		Object[] systemsData = systems.getData();
 		for (int i = 0, s = systems.size(); s > i; i++) {
-			updateEntityStates();
-
 			BaseSystem system = (BaseSystem) systemsData[i];
 			if (!system.isPassive()) {
 				system.process();
+				updateEntityStates();
```

Comment:
No change - `updateEntityStates()` is populated during system.process - if systems don't run, there can't be any entity changes to update.